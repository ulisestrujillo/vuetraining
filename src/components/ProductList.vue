<template>
  <div>
    <div> {{title}}</div>
    <ul>
      <li v-for="(product,index) in filteredProducts" :key="index">
        
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  name: 'HelloWorld',
  data () { 
    return { 
        title: 'My title',
        products:[
            {name: 'refrigerador'}
            ,{name:'productos2'}
            ,{name:'productos3'}
            ,{name:'productos4'}
            ,{name:'productos5'}
          ],
          styleObject:
            {
            "bgColor":true,
            "txtColor":true,
            "bgColorBlue":false
            }

      }
  },
  computed :{
    filteredProducts(){
      return this.products.filter((products,index) => (index >= 3));
    }
  },
  methods:{
    changeToBgBlue(){
      console.log('cambiar color azul');
      this.styleObject.bgColor = false;
      this.styleObject.bgColorBlue = true;
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<!--style src="./css/style.css" scoped-->
<style scoped>
h3 {
    margin: 40px 0 0;
  }
ul {
list-style-type: none;
padding: 0;
}
li {
border: 1px solid black;
margin: 0 10px;
}
a {
color: #42b983;
}
.bgColor{
  background-color: #42b983;
}
.bgColorBlue{
  background-color: blue;
}
.txtColor{
  color: white;
}
</style>
