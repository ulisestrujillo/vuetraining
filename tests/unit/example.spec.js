import { shallowMount } from '@vue/test-utils'
import ProductList from '@/components/ProductList.vue'

describe('ProductList.vue', () => {
  it('Matches snapshot', () => {
    const wrapper = shallowMount(ProductList)
    expect(wrapper).toMatchSnapshot()
  })
})
