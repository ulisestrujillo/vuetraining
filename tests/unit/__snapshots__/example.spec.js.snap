// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList.vue Matches snapshot 1`] = `
<div>
  <div> My title</div>
  <ul>
    <li></li>
    <li></li>
  </ul>
</div>
`;
